db/system_keyspace_clients.cc

```diff
@@ -0,0 +1,106 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/*
+ * Modified by ScyllaDB
+ * Copyright (C) 2019 ScyllaDB
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "system_keyspace_clients.hh"
+#include "system_keyspace.hh"
+#include "schema_builder.hh"
+#include "query_context.hh"
+#include "query-result-set.hh"
+#include "service/client_state.hh"
+
+namespace db::system_keyspace {
+
+extern logging::logger slogger;
+
+/// Layout based on C*-4.0.0
+schema_ptr clients() {
+    thread_local auto clients = [] {
+        schema_builder builder(make_lw_shared(schema(generate_legacy_id(NAME, CLIENTS), NAME, CLIENTS,
+            // partition key
+            {{"address", inet_addr_type},},
+            // clustering key
+            {{"port", int32_type}},
+            // regular columns
+            {
+                {"connection_stage", utf8_type},
+                {"driver_name", utf8_type},
+                {"driver_version", utf8_type},
+                {"hostname", utf8_type},
+                {"protocol_version", int32_type},
+                {"request_count", long_type},
+                {"ssl_cipher_suite", utf8_type},
+                {"ssl_enabled", boolean_type},
+                {"ssl_protocol", utf8_type},
+                {"username", utf8_type}
```

Comment:
Need to add shard here so the user knows which shard terminates the connection.

Reply:
done